### PR TITLE
Introducing ROS support for Staubli TX90 family

### PR DESCRIPTION
**Pull request by MuriloMartins**
I added support for the TX90 family as well (it is essentially a clone of the RX160, mostly renaming files and robot names within files).

As with the RX160, meshes come from the official STEP file from Staubli, joint, velocity and effort limits from the official Staubli manual, and moments of inertia were extracted from the official CAD using SolidWorks.

### CODE REVIEW

File: staubli_tx90_support/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+if (CATKIN_ENABLE_TESTING)
+  find_package(roslaunch REQUIRED)
+  roslaunch_add_file_check(tests/roslaunch_test.xml)
+endif()
+
+foreach(dir config launch meshes urdf)
+  install(
+    DIRECTORY ${dir}/
+    DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}/${dir})
+endforeach()
```
Comment (gavanderhoorn) [inline staubli_tx90_support/CMakeLists.txt]: Could you change this to the 'single line' variant (see your `_gazebo` pkg)?

**Comment by gavanderhoorn** [comment]
One minor comment, and if possible, could you decimate the `visual` meshes a bit? I'm all for quality, but 30MB for 3 variants is a bit much perhaps.

File: staubli_tx90_support/CMakeLists.txt (hunk 1 above)
Comment (MuriloMartins) [inline staubli_tx90_support/CMakeLists.txt]: The only reason why I implemented as such was for consistency with [ros-industrial/staubli/staubli_rx160_support/CMakeLists.txt](https://github.com/ros-industrial/staubli/blob/hydro-devel/staubli_rx160_support/CMakeLists.txt).

I am happy to modify (both if needed, and make a PR on [ros-industrial/staubli](https://github.com/ros-industrial/staubli)).

Comment (gavanderhoorn) [inline staubli_tx90_support/CMakeLists.txt]: I know.

We'll do the RX160 later. Please do this one now.

**Comment by MuriloMartins** [comment]
OK, I'll simplify the meshes and generate smaller files. :-)

**Comment by gavanderhoorn** [comment]
The _decimate_ modifier in Blender is a good tool for this.

File: staubli_tx90_support/CMakeLists.txt (hunk 1 above)
Comment (MuriloMartins) [inline staubli_tx90_support/CMakeLists.txt]: Done.

**Comment by gavanderhoorn** [comment]
@MuriloMartins: I hadn't noticed you updated the PR (GH doesn't send out notifications for new commits), sorry about that.

**Comment by gavanderhoorn** [comment]
@MuriloMartins: I'm confused, did you decimate the meshes in 3e491d6? A `du -csh meshes/` still shows 30MB?

**Comment by MuriloMartins** [comment]
I haven't had time to do that yet - sorry. I haven't forgotten about it.

**Comment by gavanderhoorn** [comment]
No problem at all, I was just checking, as it seemed like you had updated the PR.

**Comment by gavanderhoorn** [comment]
@MuriloMartins: I've decimated them for you. See ocadotechnology/staubli_experimental#4.

**Comment by gavanderhoorn** [comment]
@MuriloMartins: I merged this manually in 18ac6591 to add some small changes (didn't want to bother you with those).

Thanks again for the contribution, much appreciated!

**Comment by MuriloMartins** [comment]
Hi @gavanderhoorn, no worries, thanks a lot! :-)

**Pull request closed without merging**